### PR TITLE
new notificationAdapter module to encapsulate interactions with the notification service

### PR DESCRIPTION
**Pull request by techsmyth**
A pure refactoring, no functionality added or removed. Simply about maintainability of the code, and removing functionality from core/microservices module that did not belong there.

Context for this PR is that it has been clear for a while that there was a mixture of code within the microservices module that did not belong there. When starting on the notifications-143 issue it felt too risky / complicated to address as is to took the step to address the technical debt related to our interaction with the notifications service first. 

## Changes

Removed any interaction with ORM layer from the core/microservices module so that that module now only looks after subscriptions to the messaging layer.

Moved notificationsPayloadBuilder to be a service within the NotificationAdapter

Moved all knowledge of event types for Notifications within the new module

Added DTOs to make clear what is needed from the client service to send a notification event

### CODE REVIEW

File: src/services/platform/notification-adapter/dto/notification.dto.input.aspect.comment.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { IAspect } from '@domain/collaboration';
+import { IComments } from '@domain/communication/comments/comments.interface';
+import { CommunicationMessageResult } from '@domain/communication/message/communication.dto.message.result';
+import { NotificationInputBase } from './notification.dto.input.base';
+
+export class NotificationInputAspectComment extends NotificationInputBase {
```
Comment (hero101) [inline src/services/platform/notification-adapter/dto/notification.dto.input.aspect.comment.ts]: why are all DTOs classes?

File: src/services/platform/notification-adapter/notification.adapter.ts [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+  constructor(
+    @Inject(WINSTON_MODULE_NEST_PROVIDER)
+    private readonly logger: LoggerService,
+    private notificationPayloadBuilder: NotificationPayloadBuilder,
+    @Inject(NOTIFICATIONS_SERVICE) private notificationsClient: ClientProxy
+  ) {}
+
+  async calloutPublished(
+    eventData: NotificationInputCalloutPublished
+  ): Promise<boolean> {
```
Comment (hero101) [inline src/services/platform/notification-adapter/notification.adapter.ts]: when does it return `false`?
Maybe `void` is more suitable when the same result is always returned.

File: src/services/platform/notification-adapter/notification.adapter.ts [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+@Injectable()
+export class NotificationAdapter {
+  constructor(
+    @Inject(WINSTON_MODULE_NEST_PROVIDER)
+    private readonly logger: LoggerService,
+    private notificationPayloadBuilder: NotificationPayloadBuilder,
+    @Inject(NOTIFICATIONS_SERVICE) private notificationsClient: ClientProxy
+  ) {}
+
+  async calloutPublished(
```
Comment (hero101) [inline src/services/platform/notification-adapter/notification.adapter.ts]: At an SSE meeting, we agreed to include all the access modifiers so the code is less error-prone and increases visibility.

**Review by hero101 — CHANGES_REQUESTED** [review]
- There are failing tests.

File: src/services/platform/notification-adapter/notification.adapter.ts (hunk 3 above)
Comment (techsmyth) [reply src/services/platform/notification-adapter/notification.adapter.ts]: fixed

File: src/services/platform/notification-adapter/notification.adapter.ts (hunk 2 above)
Comment (techsmyth) [reply src/services/platform/notification-adapter/notification.adapter.ts]: updated to be void

File: src/services/platform/notification-adapter/dto/notification.dto.input.aspect.comment.ts (hunk 1 above)
Comment (techsmyth) [reply src/services/platform/notification-adapter/dto/notification.dto.input.aspect.comment.ts]: converted to interfaces

**Review by hero101 — APPROVED** [review]

**Pull request merged**